setup.py

```diff
@@ -41,7 +41,8 @@ def get_version(rel_path: str) -> str:
     "treelib~=1.6.4",
     "pathvalidate~=3.0.0",
     "python-dateutil",
-    "tenacity~=8.2.3"
+    "tenacity~=8.2.3",
+    "boto3~=1.34.54"
```

Comment:
Still not sure if we should pin the versions or not, considering we want to play nice with other libraries that get installed, and A LOT of libraries install boto. Might break pip install for dependencies.
The solution I used before that was to hide the boto behind a lazy import, so the user gets an exception to `pip install boto`. Used that for the bucket clients and for the fiftyone imports.
However this feature will probably be used more often than those... I'll mull over what exactly to do here

Reply:
Reading up on this: https://discuss.python.org/t/should-i-be-pinning-my-dependencies/13159